pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -209,6 +209,21 @@ func (e *endpointsController) buildIstioEndpoints(endpoint interface{}, host hos
 			// EDS and ServiceEntry use name for service port - ADS will need to map to numbers.
 			for _, port := range ss.Ports {
 				istioEndpoint := builder.buildIstioEndpoint(ea.IP, port.Port, port.Name, discoverabilityPolicy)
+				istioEndpoint.HealthStatus = model.Healthy
+				endpoints = append(endpoints, istioEndpoint)
+			}
+		}
+		for _, ea := range ss.NotReadyAddresses {
+			pod, expectedPod := getPod(e.c, ea.IP, &metav1.ObjectMeta{Name: ep.Name, Namespace: ep.Namespace}, ea.TargetRef, host)
+			if pod == nil && expectedPod {
+				continue
+			}
+			builder := NewEndpointBuilder(e.c, pod)
+
+			// EDS and ServiceEntry use name for service port - ADS will need to map to numbers.
+			for _, port := range ss.Ports {
+				istioEndpoint := builder.buildIstioEndpoint(ea.IP, port.Port, port.Name, discoverabilityPolicy)
+				istioEndpoint.HealthStatus = model.UnHealthy
```

Comment:
Looks this is duplicate with above , and can be abstracted. 

I am thinking maybe we should have a feature flag for it.